### ISSUE TITLE
Issue with Plot label map

### ISSUE THREAD

**Issue by nepersica** [fepegar/torchio#730]
**Problem**
Thank you for provide great library:)
I have a problem about plotting LabelMap using plot().

**To reproduce**

```python
def get_data(args):
    print(f'> Getting {args.mode} data.... > Augmentation mode: {args.transform}')
    data_path = os.path.join(args.data_dir, args.hospital)
    
    dataset = MedicalDataset(root_dir=data_path, args=args)
    for data in dataset.training_set:
        one_subject = data
        label = data['label'][tio.DATA]
        print((data['label'][tio.DATA]== 1).nonzero(as_tuple=True))
        one_subject.plot()
    return dataset

class MedicalDataset(Dataset):
    def __init__(self, root_dir, args):
        self.root_dir = root_dir
        self.image_size = args.image_size

        self.subjects = []
        dicom_list = os.listdir(os.path.join(self.root_dir, 'dicom'))
        for dicom_dir in tqdm(dicom_list):
            
            subject = tio.Subject(
                input=tio.ScalarImage(os.path.join(root_dir, 'dicom', dicom_dir)),
                label=tio.LabelMap(os.path.join(root_dir, 'label', dicom_dir+'.nii.gz')),
            )
            self.subjects.append(subject)

        self.training_set = tio.SubjectsDataset(self.subjects)
        print(f'{args.mode} dataset_size:', len(self.training_set), 'subjects')
```

**Expected behavior**
Large size of segment is shown on plot but, small size of segment is not shown like "Actual behavior.
So I check label category on nii.gz file by printing tensor, but there was no probelm.
Does it originally working like this?

![image](https://user-images.githubusercontent.com/45097022/140635147-47bf0ece-53e4-4b84-bb2f-3a876ed7e15b.png)

**Actual behavior**

![image](https://user-images.githubusercontent.com/45097022/140635137-b4730bea-de00-46f4-b324-06fd25509fc3.png)

**System info**

Output of `python <(curl -s https://raw.githubusercontent.com/fepegar/torchio/main/print_system.py)`:

```
Platform:	Windows-10-10.0.19043-SP0
torchio:	0.18.62
torch:	1.9.1+cu111
numpy:	1.21.2
Python:	3.9.6 | packaged by conda-forge | (default, Jul 11 2021, 03:37:25) [MSC v.1916 64 bit (AMD64)]
```

**Comment by fepegar** [comment]
Hi, @nepersica. Thanks for reporting and for your kind words. Could you please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) that I can run on my computer? That way, it will be easier for me to help you.

**Comment by nepersica** [comment]
I appreciate you responding so quickly to my issue☺

My codes consist of 'main.py', 'train.py',  'dataloader.py'

## main.py
```
import argparse
import os
os.environ['KMP_DUPLICATE_LIB_OK']='True'
import torch
from train import train
from test import test
import torch.nn as nn

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')

def get_args():
    parser = argparse.ArgumentParser(description="Train the UNet",
                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)

    parser.add_argument("--lr", default=1e-4, type=float, dest="lr")
    parser.add_argument("--batch_size", default=1, type=int, dest="batch_size")
    parser.add_argument("--num_epoch", default=100, type=int, dest="num_epoch")

    parser.add_argument("--transform", default="on", type=str, dest="transform")
    parser.add_argument("--image_size", default=256, type=int, dest="image_size")

    parser.add_argument("--gamma", default=3, type=float, dest="gamma")
    parser.add_argument("--rho", default=0.5, type=float, dest="rho")
    parser.add_argument("--momentum", default=0.9, type=float, help="SGD Momentum.")
    parser.add_argument("--step_size", default=20, type=float, dest="step_size")
    parser.add_argument("--optimizer", default='adam', type=str)

    parser.add_argument("--data_dir", default="./dicom_dataset", type=str, dest="data_dir")
    parser.add_argument("--result_dir", default="./result", type=str, dest="result_dir")
    parser.add_argument("--log_dir", default="./log", type=str, dest="log_dir")
    parser.add_argument("--num_workers", default=4, type=int, dest="num_workers")

    parser.add_argument("--mode", default="train", type=str, dest="mode")
    parser.add_argument("--train_continue", default="off", type=str, dest="train_continue")
    parser.add_argument("--pretrained", default="off", type=str, dest="pretrained")
    parser.add_argument("--parser_mode", default="hdf5", type=str, dest="parser_mode")

    
    
    parser.add_argument("--model_dimension", default="3d", type=str)
    parser.add_argument("--model_name", default="U-Net", type=str)

    parser.add_argument("--hospital", default="HALLYM",  type=str, dest="hospital")

    args = parser.parse_args()

    return args

def main(args):
    if args.mode == 'train':  
        train(args, device)

if __name__ == '__main__':
    args = get_args()
    main(args)
```

## train.py
```
from dataloader import get_data

def train(args, device):
    train_dataset = get_data(args)
```

## dataloader.py
```
from glob import glob
import os
import sys
sys.path.append('./')
import torch
# from medpy.io import load
import numpy as np
import torch.nn.functional as F
import torchio as tio
from torchio import AFFINE, DATA
import torchio
from torchio import ScalarImage, LabelMap, Subject, SubjectsDataset, Queue
from torchio.data import UniformSampler
from torchio.transforms import (RandomFlip, RandomAffine, RandomElasticDeformation, RandomNoise, RandomMotion
    , RandomBiasField, RescaleIntensity, Resample, ToCanonical, ZNormalization, CropOrPad, HistogramStandardization
    , OneOf, Compose,)
from pathlib import Path
import config as cfg

from torch.utils.data import Dataset
from torch.utils.data import DataLoader

from dataparser import get_data_list
import cv2 as cv
from tqdm import tqdm
import matplotlib.pyplot as plt

from  visualization import *

def get_data(args):
    print(f'> Getting {args.mode} data.... > Augmentation mode: {args.transform}')
    data_path = os.path.join(args.data_dir, args.hospital)
    
    dataset = MedicalDataset(root_dir=data_path, args=args)
    # for data in dataset.training_set:
    #     one_subject = data
    #     print(one_subject)
    #     print(one_subject.input)
    #     print(one_subject.label)

    # dataset.training_set

def show_image(data):
    size_r, size_a, size_s = data.shape
    for k in range(size_s):
        slice = data[:, :, k]# or just data[..., k]
        plt.imshow(slice)
        plt.show()

class MedicalDataset(Dataset):
    def __init__(self, root_dir, args):
        self.root_dir = root_dir
        self.image_size = args.image_size

        self.subjects = []
        dicom_list = os.listdir(os.path.join(self.root_dir, 'dicom'))
        for dicom_dir in tqdm(dicom_list):
            
            subject = tio.Subject(
                input=tio.ScalarImage(os.path.join(root_dir, 'dicom', dicom_dir)),
                label=tio.LabelMap(os.path.join(root_dir, 'label', dicom_dir+'.nii.gz')),
            )
            self.subjects.append(subject)
        
        if args.mode == 'train':
            self.transform = Compose([
                                    ToCanonical(),
                                    Resample('label'),
                                    CropOrPad((128, 128, 32)),
                                    # RandomMotion(),
                                    RandomBiasField(),
                                    ZNormalization(),
                                    # RandomNoise(),
                                    RandomFlip(axes=(0,)),
                                    # OneOf({
                                    #     RandomAffine(): 0.8,
                                    #     RandomElasticDeformation(): 0.2,
                                    # }),
                                    ])
            # self.dataset = tio.SubjectsDataset(self.subjects, transform=self.transform)
            self.dataset = tio.SubjectsDataset(self.subjects)
        print(f'{args.mode} dataset_size:', len(self.dataset), 'subjects')

        for idx, data in enumerate(self.dataset):
            one_subject = data
            t1 = one_subject.input
            t2 = one_subject.label
            print(one_subject)
            print(one_subject.input)
            print(one_subject.label)
            plot_subject(one_subject, output_path=f'./crop/{idx}_origin.png')

            subject = tio.Subject(input=t1, label=t2)
            transforms = tio.ToCanonical(), tio.Resample('label'), CropOrPad((128, 128, 32))
            transform = tio.Compose(transforms)

            fixed = transform(subject)
            plot_subject(fixed, output_path=f'./crop/{idx}_fixed.png')

        patch_size = cfg.patch_size_3d
        queue_length = 300
        samples_per_volume = 5
        sampler = tio.data.UniformSampler(patch_size)

        if args.mode == 'train':
            self.queue_dataset = Queue(
                subjects_dataset=self.dataset,
                max_length=queue_length,
                samples_per_volume=samples_per_volume,
                sampler=sampler,
            )
            data_loader_patches = DataLoader(self.queue_dataset, batch_size=args.batch_size, shuffle=True)
    
    def __len__():
        return len(dataset)

    def _to_tensor(self, image):        
        np_image = np.array(image)
        np_image = np.transpose(np_image, (1,2,0))
        np_image = np_image[np.newaxis, ...]
        return torch.from_numpy(np_image)
    

    def _get_image(self, image, image_type):
        image = np.array(image)
        image = image.astype('float32')

        return image

if __name__ == '__main__':
    from main import get_args
    args = get_args()

    train_data, train_loader = get_data(args)
```

**Comment by fepegar** [comment]
Please read the link I shared above (https://github.com/fepegar/torchio/issues/730#issuecomment-962593372) and provide a _minimal and reproducible_ example.

**Comment by nepersica** [comment]
I’m not good at English, so misunderstood the link. Very sorry about that.😭

1. I load image(dicom file) and label(nifti file) to ScalarIamge and LabelMap, respectively.
```
  subject = tio.Subject(
      input=tio.ScalarImage(os.path.join(root_dir, 'dicom', dicom_dir)),
      label=tio.LabelMap(os.path.join(root_dir, 'label', dicom_dir+'.nii.gz')),
  )
  self.subjects.append(subject)
  self.training_set = tio.SubjectsDataset(self.subjects)
```

2. And want to check subjects loaded well,  print the subject by plot().
```
    for data in dataset.training_set:
        one_subject = data
        label = data['label'][tio.DATA]
        one_subject.plot()
    return dataset
```
As a result, Image was showed well, but label was not.
In 10 subject examples, only 1 subjectloaded on the plot.

I print that the label has value of 1 by under code, and check the values are located well .
```
print((data['label'][tio.DATA]== 1).nonzero(as_tuple=True))
```

At first, I thought the plot() function has problem.
So, I test with Decathlon dataset, and cheeck image and label were showed well.
![image](https://user-images.githubusercontent.com/45097022/140754319-59b8c9f0-a76f-4810-9a07-078c42a58113.png)

I think.. it might be the order of my nifti are inversed, but not sure..:(

**Comment by nepersica** [comment]
I think just label(nifti file) is okay, you can download them below link!
https://drive.google.com/drive/folders/1Fy1jAbeszKmuYTi5WAT6FoV__UduBL_e?usp=sharing

**Comment by fepegar** [comment]
So those files are ok? Can you share one that is not?

**Comment by nepersica** [comment]
Yes, just nifti file is  okay. Thank you for your concern. But, just in case, I'm going to delete the link after you download it.

**Comment by fepegar** [comment]
If you say these files are ok, I will not download them. If you have issues with a specific one, please let me know and I will try to figure out what's wrong.

**Comment by nepersica** [comment]
Oh, I misuderstood what you mean again😅I understand is it okay to share files.

Above link is the trouble datasets. very sorry..

**Comment by fepegar** [comment]
Alright. You discovered a bug. Thanks for reporting and for sharing your data! I will fix it in a second.

**Comment by fepegar** [comment]
@all-contributors please add @nepersica for bug

**Comment by fepegar** [comment]
Fixed in `v0.18.63`.

**Comment by nepersica** [comment]
Thank you for  your kind answering and fixing problem even though there was a problem talking with me.
I wish you every happiness!!☺🌹

**Comment by fepegar** [comment]
No worries!
